library/src/main/java/com/novoda/downloadmanager/lib/DownloadManager.java

```diff
@@ -449,15 +449,25 @@ public int markRowDeleted(long... ids) {
     }
 
     /**
-     * Cancel downloads and remove them from the download manager.  Each download will be stopped if
+     * Cancel batch downloads and remove them from the download manager.  Each download will be stopped if
      * it was running, and it will no longer be accessible through the download manager.
-     * If there is a downloaded file, partial or complete, it is deleted.
+     * If there are any downloaded files, partial or complete, they will be deleted.
      *
-     * @param ids the IDs of the downloads to remove
-     * @return the number of downloads actually removed
+     * @param ids the IDs of the batches to remove
+     * @return the number of batches actually removed
      */
-    public int remove(long... ids) {
-        return markRowDeleted(ids);
+    public int removeBatches(long... ids) {
+        if (ids == null || ids.length == 0) {
+            throw new IllegalArgumentException("called with nothing to remove. input param 'ids' can't be null");
+        }
+        ContentValues values = new ContentValues();
+        values.put(Downloads.Impl.Batches.COLUMN_DELETED, 1);
+        // if only one id is passed in, then include it in the uri itself.
+        // this will eliminate a full database scan in the download service.
+        if (ids.length == 1) {
+            return mResolver.update(ContentUris.withAppendedId(Downloads.Impl.BATCH_CONTENT_URI, ids[0]), values, null, null);
```

Comment:
It is, not our great idea though: http://androidxref.com/4.3_r2.1/xref/frameworks/base/core/java/android/app/DownloadManager.java#955